### DISCUSSION TITLE
Auto-sync with navigator does not work with Oracle

### DISCUSSION THREAD

**Discussion by jyinpci** [dbeaver/dbeaver#12175] [Q&A] [answered]
Once a database A is selected in the navigator, all open tabs of SQL editors should be automatically connected to that database A.  This is how it works with Postgresql.  But with Oracle, if multiple SQL editors tabs are opened, when database A is selected in the navigator, only the currently selected SQL editor tab is connected to the database A.  All other SQL editor tabs stay with their previous database.  After switching to another SQL editor tab, one would need to click on the database A in navigator again to change the database for the newly selected SQL editor tab.

I have a number of sql scripts, opened in separate SQL editor tabs, that I use to perform various tasks when I work with a particular database.  I just need to select the SQL editor tab and execute it to perform a task.  When I switch database A to database B in the navigator, I would like all opened SQL editor tabs to switch to database B.  Otherwise, I may perform a task on a wrong database.  Any help will be appreciated greatly.

I have been seeing this in v7.  I am now using 7.3.5 community edition with Oracle Express on Windows 10 Pro.

**Comment by kseniiaguzeeva** [comment]
Most probably **Open separate connection for each editor** is enabled in Oracle together with Auto-sync connection with navigator. Try to disable it.
![Screenshot 2021-02-18 at 14 01 48](https://user-images.githubusercontent.com/51405061/108347874-20212700-71f2-11eb-835c-cff44f8e44db.png)

**Comment by jyinpci** [comment]
Kseniia,
Thank you for your prompt response.  I did have the “Open separate connection for each editor” unchecked”.  Could it be overridden by some other settings related to Oracle?  I never had this issue with Postgresql until I added Oracle.
BTW, I am using the Community edition.

**Comment by kseniiaguzeeva** [comment]
Is it possible to send a video with the issue?

**Comment by jyinpci** [comment]
Hi Kseniia,

 

Thank you for your follow-up.  I really should have recorded a video and have tried, but my database names all contain names of my customer companies, they might not be pleased to have such videos uploaded to a public site.

 

I did experiment with different configurations and have more to report.  I think having multiple connections (to different databases) in the navigator seems to cause the failure of sync connection between navigator and SQL editor.  In the configuration below, the navigator has two connections to two databases (one postgres, one oracle – but having two Oracle connections also breaks the sync).  I did not expand the connection to show schemas because of the name sensitivity.

 



I found out that if there is only one database connection in the navigator, e.g., if only one of the postgres connection or the Oracle connection, the SQL editor connection would sync up with the navigator.  Once both connections are added to the navigator, the SQL editor connection stops to sync up with navigator.   In the past, I had only the postgres connection and it stopped sync-ing after I added Oracle.  So, I thought it has something to do with Oracle.  But after I deleted the postgres connection, then the Oracle connection would sync fine between navigator and SQL editor.  In fact, if I add another Oracle connection (to a different Oracle database), the sync would be broken.

 

I hope this helps.

**Comment by kseniiaguzeeva** [comment, marked as answer]
Sorry for the long answer. I clarified how the option works. Seems, now I understand your case. First of all it was wrong advice to enable _Open separate connection for each editor_. Please disable it for your case and enable **Auto-sync connection with Navigator**
![dbeaver_eLaB4A4TSh](https://user-images.githubusercontent.com/51405061/111484915-3b2a7c80-8747-11eb-934f-6e998bed855e.png)
How it works:
1. When you have one connection with several schemas in DB Navigator and several SQL Editor windows opened: you click a schema in DB Navigator and then all SQL Editor windows change schemas on the selected one.
2. When you have two connections (e.g. PostgreSQL and Oracle) in DB Navigator and several SQL Editor windows for both connections: when you click a schema (e.g. in Oracle connection) in DB Navigator, SQL Editor windows for this connection only (Oracle in my example) change the schemas on the selected one. 
If you want to synchronise SQL Editor windows of one connection  (PostgreSQL in my example) with different connection in DB Navigator (Oracle in my example), you have to do it for every SQL Editor window separately (open SQL Editor window, then click a schema in DB Navigator, then open other SQL Editor window etc.)

https://user-images.githubusercontent.com/51405061/111486652-b80a2600-8748-11eb-9e0b-839df5df177b.mp4

**Comment by nacesprin** [comment]
What auto-sync with navigator does exactly? I dont know what is the goal of this functionality.
Thanks

**Reply by jyinpci** [reply]
This is a very useful and important function when one needs to work with one set of scripts on multiple databases.  For example, I often need to look at both my test database and production database.  I have one set of scripts that I use for different tables in the database.  The auto-sync with navigator function allows me to make a switch in the navigator from one database to another, then all scripts opened in the editor will automatically switched to the connection to that database shown in the navigator.   This helps to avoid looking/updating wrong database by mistake.

Unfortunately, this function still does not work well with Oracle (it only works when switching schemas within one database, but not different databases).  I wish DBeaver can come up with a good solution for this, i.e. allow the user to easily switch the connection of all the open scripts to a new database.  Apparently, this seems to be a non-trivial thing to do.
